File: claimshield-ai/src/evidence_check.py
def check_completeness(claim, required_docs):
    documents = claim.get("documents", {})

    present_docs = []
    missing_docs = []

    for doc in required_docs:
        document = documents.get(doc, {})
        if document.get("present") is True:
            present_docs.append(doc)
        else:
            missing_docs.append(doc)

    total_required = len(required_docs)
    present_count = len(present_docs)

    if total_required == 0:
        completeness_pct = 100
    else:
        completeness_pct = round((present_count / total_required) * 100)

    return {
        "isComplete": len(missing_docs) == 0,
        "completenessPct": completeness_pct,
        "requiredDocuments": required_docs,
        "presentDocuments": present_docs,
        "missingDocuments": missing_docs,
    }

File: claimshield-ai/server/lib/evidenceCheck.js
// DETERMINISTIC LAYER — no AI involved.
// Decides only what evidence is present vs missing, based purely on the
// claim's `documents` map. Never infers or assumes a document exists.

const DOC_LABELS = {
  claimForm: "Claim Form",
  incidentDescription: "Incident Description",
  repairEstimate: "Repair Estimate",
  fir: "FIR / Police Report",
};

const CORE_DOCS = ["claimForm", "incidentDescription", "repairEstimate", "fir"];

export function checkCompleteness(claim, requiredDocs) {
  // requiredDocs: docs required for whichever policy clauses could apply.
  // Falls back to the four core document types when not specified.
  const relevant = requiredDocs && requiredDocs.length ? requiredDocs : CORE_DOCS;

  const available = [];
  const missing = [];

  for (const key of relevant) {
    const doc = claim.documents[key];
    if (doc && doc.present) {
      available.push({ key, label: DOC_LABELS[key] });
    } else {
      missing.push({ key, label: DOC_LABELS[key] });
    }
  }

  const totalConsidered = relevant.length;
  const completenessPct = totalConsidered
    ? Math.round((available.length / totalConsidered) * 100)
    : 100;

  return {
    completenessPct,
    available,
    missing,
    isComplete: missing.length === 0,
  };
}
